### PR TITLE
feat: add levenshtein implementation

### PR DESCRIPTION
**Pull request by fadilaaaa**
this algo gives O(m*n) complexity

issue #36

### CODE REVIEW

**Comment by WahidinAji** [comment]
please, update the usage in the readme.md :D

**Comment by fadilaaaa** [comment]
> please, update the usage in the readme.md :D

done sir (●'◡'●)

**Comment by aldy505** [comment]
@fadilaaaa there's a merge conflict

File: go/levenshtein.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+package pehape
+
+// LevenshteinDistance calculate the distance between two string
```
Comment (aldy505) [inline go/levenshtein.go:3]:
```suggestion
// Levenshtein calculates the distance between two strings.
```

File: go/readme.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+ //result : 3
+ ```
+ with custom cost
+ ```go
+ str1 := "kitten"
+ str2 := "sitting"
+ insertion_cost := 1
+ replacement_cost := 2
+ deletion_cost := 3
+ fmt.Println(pehape.Levenshtein(str1,str2,insertion_cost,replacement_cost,deletion_cost))
```
Comment (aldy505) [inline go/readme.md:19]: please format the code

File: go/readme.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
 `import "github.com/teknologi-umum/pehape/go"`
 
 ## Usage?
+ * `Levenshtein`
+ ```go
+ str1 := "kitten"
+ str2 := "sitting"
+ fmt.Println(pehape.Levenshtein(str1,str2))
```
Comment (aldy505) [inline go/readme.md:9]:
```suggestion
 fmt.Println(pehape.Levenshtein(str1, str2))
```

File: go/readme.md [hunk 4]
```diff
@@ -2,6 +2,24 @@
...
+ //result : 3
+ ```
+ with custom cost
+ ```go
+ str1 := "kitten"
+ str2 := "sitting"
+ insertion_cost := 1
+ replacement_cost := 2
+ deletion_cost := 3
```
Comment (aldy505) [inline go/readme.md:18]: don't use snake_case. use camelCase instead

**Review by aldy505 — CHANGES_REQUESTED** [review]

**Review by aldy505 — APPROVED** [review]

**Pull request merged**